### PR TITLE
Add multiple messages

### PR DESCRIPTION
**Pull request by illiriks**
- Multiple messages can now be sent with one API call.
- Fixed `nock` in test for single message so the test actually tests
- Unbroke Travis (I accidentally deleted the `npm test` line when adding the autodeploy script)

### CODE REVIEW

**Comment by dtolb** [comment]
@illiriks per our convo earlier, go ahead and add some examples for sending messages. I'd be nice if one of the examples (use multiple @example tags) shows how to process the result.

**Comment by dtolb** [comment]
@illiriks define the new message response for array. And actually, I just realized the function could return two different types depending on input... I don't know if that's best practice. Granted the input types are different, so there _should_ be _some_ self-awareness. 

I'm almost thinking `message.sendAll` to make the return type consistent for functions. 

@fuchsnj @a-type @nguyer @nbragdon thoughts?

**Comment by illiriks** [comment]
@dtolb You're right; we either get an array of things that aren't quite messages or a single message, depending on what you give it. I can split it into multiple functions pretty easily. I wanted to hold off on writing examples till a consensus was reached on how the results should be formatted. My instinct, if we were to split it into a `sendAll` type thing is to make the function behave similarly to `map`ping `send` on an array (with better performance, of course)

File: lib/message.js [hunk 1]
```diff
@@ -28,7 +28,28 @@ var Message = function (client) {
 	 * @param  {String} [params.receiptRequested=none] Requested receipt option for outbound messages:
 	 * `none` `all` `error`
 	 * @param  {Function} [callback] A callback for the new message object
-	 * @return {MessageResponse} A promise for the new message object
+	 * @returns {MessageResponse} A promise for the new message object
+	 * @example
+	 * var CatapultClient = require('node-bandwidth');
```
Comment (dtolb) [inline lib/message.js]: take this part out, we don't need to show them how to create the client.

File: lib/message.js [hunk 2]
```diff
@@ -28,7 +28,28 @@ var Message = function (client) {
...
+	 * client = new CatapultClient({
+	 *   userId    : $YOUR_USER_ID,
+	 *   apiToken  : $YOUR_API_TOKEN,
+	 *   apiSecret : $YOUR_API_SECRET
+	 * });
+	 * client.Message.send({
```
Comment (dtolb) [inline lib/message.js:33]: we just want this :)

File: lib/message.js [hunk 3]
```diff
@@ -39,12 +60,118 @@ var Message = function (client) {
...
+	 * @param  {String} [params.fallbackUrl] The server URL used to send message events
+	 * if the request to callbackUrl fails
+	 * @param  {String} [params.tag] A string that will be included in the callback events of the message
+	 * @param  {String} [params.receiptRequested=none] Requested receipt option for outbound messages:
+	 * `none` `all` `error`
+	 * @param  {Function} [callback] A callback for the array of ExtendedMessageResponse
+	 * @returns [ExtendedMessageResponse] A promise for the array of ExtendedMessageResponse
+	 * @example
+	 * var CatapultClient = require('node-bandwidth');
+	 * client = new CatapultClient({
```
Comment (dtolb) [inline lib/message.js]: same as above

File: lib/client.js [hunk 4]
```diff
@@ -13,7 +13,7 @@ var Client = function (config) {
 	}
 
 	var handleResponse = function (response) {
-		if (response.statusCode !== 200 && response.statusCode !== 201) {
+		if (response.statusCode !== 200 && response.statusCode !== 201 && response.statusCode !== 202) {
```
Comment (dtolb) [inline lib/client.js:16]: add test for this. Make sure that we don't `error` on 200, 201, 202 and that we do `error` on other things...

File: lib/message.js [hunk 5]
```diff
@@ -39,12 +54,112 @@ var Message = function (client) {
 		.then(function (response) {
 			var message = params;
 			var location = response.headers.location;
-			var messageId = location.substring(location.lastIndexOf("/") + 1);
-			message.id = messageId;
+			message.id = location.substring(location.lastIndexOf("/") + 1);
```
Comment (dtolb) [inline lib/message.js:57]: yeah, ok

File: lib/message.js [hunk 6]
```diff
@@ -39,12 +54,112 @@ var Message = function (client) {
...
+	 * outgoing message will be sent
+	 * @param  {Number} [params.callbackTimeout] Determine how long should the platform wait for
+	 * callbackUrl"s response before timing out (milliseconds)
+	 * @param  {String} [params.fallbackUrl] The server URL used to send message events
+	 * if the request to callbackUrl fails
+	 * @param  {String} [params.tag] A string that will be included in the callback events of the message
+	 * @param  {String} [params.receiptRequested=none] Requested receipt option for outbound messages:
+	 * `none` `all` `error`
+	 * @param  {Function} [callback] A callback for the array of ExtendedMessageResponse
+	 * @returns [ExtendedMessageResponse] A promise for the array of ExtendedMessageResponse
```
Comment (dtolb) [inline lib/message.js]: Should be `{ExtendedMessageResponse}` the array type is handled in the `@class` definition below... ideally we'd want to use `@typedef` but the `@typedef` doesn't populate the global namespace by default with our jsdoc tooling, so there is not clear way to access that return type w/o clicking around.

See [applications](https://github.com/bandwidthcom/node-bandwidth/blob/master/lib/application.js#L17) for how this works

File: lib/message.js [hunk 7]
```diff
@@ -121,6 +236,25 @@ var Message = function (client) {
 module.exports = Message;
 
 /**
+ * @class ExtendedMessageResponse
```
Comment (dtolb) [inline lib/message.js:239]: See how [applications](https://github.com/bandwidthcom/node-bandwidth/blob/master/lib/application.js#L260) handle this to build out the proper jsdoc responses

**Pull request merged**